service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/AdditionalRouter.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.javadsl.server
+
+import java.util
+
+import play.api.inject.Injector
+import play.routing.Router
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable
+
+trait AdditionalRouter {
+  def prefix: Option[String]
+  def withPrefix(path: String): AdditionalRouter
+}
+
+final case class ClassBased[R <: Router](classType: Class[R], prefix: Option[String]) extends AdditionalRouter {
+
+  def this(classType: Class[R]) = this(classType, None)
+
+  def withPrefix(path: String): ClassBased[R] = copy(prefix = Option(path))
+}
+
+final case class InstanceBased(router: Router, prefix: Option[String]) extends AdditionalRouter {
+
+  def this(classType: Router) = this(classType, None)
+
+  def withPrefix(path: String): InstanceBased = copy(prefix = Option(path))
+}
```

Comment:
As explained on the call, we are here trying to emulate Play's behaviour. 

The above example would result in a `Router` defined `/sufix/infix/prefix`. A little bit strange, but that's how Play works.

I will adapt accordingly because current impl is replacing the prefix each time.